src/LayerEditor/ui/data/interpolated_node_set_item.py

```diff
@@ -13,7 +15,31 @@ def __init__(self, itf_node_set1, itf_node_set2, interface):
         self.interface = interface
         """Interface index"""
 
+    def is_top_and_bottom_equal(self):
+        return self.top_itf_node_set == self.bottom_itf_node_set
+
+    def get_shapes(self):
+        """Topology must be the same so shape should be too"""
+        return self.top_itf_node_set.decomposition.decomp.shapes
+
+    @property
+    def block(self):
+        return self.top_itf_node_set.decomposition.block
+
     def save(self):
-        return InterpolatedNodeSet(dict(surf_nodesets=[self.top_itf_node_set.save(),
-                                                       self.bottom_itf_node_set.save()],
-                                        interface_id=self.interface.index))
\ No newline at end of file
+        return InterpolatedNodeSet(dict(surf_nodesets=(self.top_itf_node_set.save(),
+                                                       self.bottom_itf_node_set.save()),
+                                        interface_id=self.interface.index))
+
+    @undo.undoable
+    def change_decomposition(self, top_decomp, bot_decomp):
+        self.top_itf_node_set.change_decomposition(top_decomp)
+        self.bottom_itf_node_set.change_decomposition(bot_decomp)
+
+    def __eq__(self, other):
```

Comment:
It is used for finding whether instances of this object have the same attributes. I do not see how it could be insufficient for some usage. It is just potentially slow.